torch_glow/src/PyTorchModelLoader.cpp

```diff
@@ -1121,6 +1174,200 @@ glow::NodeValue PyTorchModelLoader::rescaleIntToUint(glow::NodeValue input) {
+
+  int32_t outOffset;
+  ASSIGN_VALUE_OR_RETURN_ERR(
+      outOffset, iValToInt(getGlowIValueForValue(
+                     inputs[QuantizedConv2dReluInputs::zero_point])));
+
+  // calc output type
+  std::vector<glow::unsigned_t> kernels = {
+      static_cast<glow::unsigned_t>(weightShape.h),
+      static_cast<glow::unsigned_t>(weightShape.w)};
+  auto outSz = glow::calculateConvPoolOutputDims(
+      inputShape.h, inputShape.w, kernels, strides, pads, dilation);
+  std::array<size_t, 4> outDims = {
+      {input.dims()[0], outSz.first, outSz.second, weightShape.n}};
+  glow::TypeRef outTy = F_.getParent()->uniqueType(
+      glow::ElemKind::Int8QTy, outDims, outScale, outOffset);
+
+  if (isGroupwiseQuantized) {
+    RETURN_ERR_IF_NOT(dilation <= 1,
+                      "Dilation not supported for group quantized convolution");
+
+    // Get groupwise quantized params if needed
+    glow::Tensor wScalesTensor, wOffsetsTensor;
+    glow::Constant *wScales, *wOffsets;
+
+    // extract qparams from ptWeightTensor.
+    // Notice since the memory of qparams may not be continous
+    // we CANNOT use the data ptr of this chunk of memory and
+    // convert them into glow tensor directly by using PtTensorToGlowTensor.
+    // Instead, we extract them one after one.
+    std::vector<float> scalesVector;
+    std::vector<int32_t> offsetsVector;
+    std::vector<size_t> dims;
+    const int n = ptWeightTensor.q_per_channel_scales().size(0);
+    dims.push_back(n);
+    for (int i = 0; i < n; i++) {
+      float scale =
+          ptWeightTensor.q_per_channel_scales().to(at::kFloat)[i].item<float>();
+      int32_t offset = ptWeightTensor.q_per_channel_zero_points()
+                           .to(at::kInt)[i]
+                           .item<int32_t>();
+      scalesVector.push_back(scale);
+      offsetsVector.push_back(offset);
+    }
+
+    // construct qparam constants
+    auto scaleType = glow::Type(ElemKind::FloatTy, dims);
+    auto offsetType = glow::Type(ElemKind::Int32ITy, dims);
+    wScalesTensor = glow::Tensor(scalesVector.data(), &scaleType);
+    wOffsetsTensor = glow::Tensor(offsetsVector.data(), &offsetType);
+
+    wScales = F_.getParent()->createConstant("channel_wised_scales_of_qconv",
+                                             std::move(wScalesTensor));
+    wScales->ensureIsOwned();
+    wOffsets = F_.getParent()->createConstant("channel_wised_offsets_of_qconv",
+                                              std::move(wOffsetsTensor));
+    wOffsets->ensureIsOwned();
+
+    auto qconv = F_.createChannelwiseQuantizedConv(
+        "qconv_channel_wised", input, weightConstant, biasConstant, wScales,
+        wOffsets, outTy, kernels, strides, pads, groups);
+    if (isRelu) {
+      glow::ReluNode *qrelu = F_.createRELU("qconv_relu", qconv);
+      output = F_.createTranspose("channel_wised_qconv_relu_output_transposed",
+                                  qrelu->getResult(), NHWC2NCHW);
+    } else {
```

Comment:
you can just have transpose as last thing here and remove this else

Reply:
@jackm321 No I cannot actually, since one is qrelu and one is qconv, they have different type.

Reply:
You can just use their output NodeValue